### ISSUE TITLE
Question on permissions

### ISSUE THREAD

**Issue by IzzySoft** [dreautall/waterfly-iii#295]
During the past few weeks, my APK scanner got some additional checks. So for today's release it reported:

    ! repo/com.dreautall.waterflyiii_2373.apk declares sensitive permission(s): android.permission.READ_EXTERNAL_STORAGE android.permission.READ_MEDIA_IMAGES android.permission.READ_MEDIA_VIDEO android.permission.READ_MEDIA_AUDIO

Could you please clarify what those permissions are needed for? I couldn't find out by the apps description or its README. Thanks in advance!

**Comment by IzzySoft** [comment]
Thanks for the fast response! Yes, that screenshot was not in Fastlane, so I didn't see it before. Added the permissions to the allow-list then:

![image](https://github.com/dreautall/waterfly-iii/assets/6781438/37a2a4c3-15f3-4d7d-aa9c-3108900c5780)

(if ou want to have explanations added for the other permissions, let me know and I can do it. I just asked for the "sensitive ones" which otherwise would show up with "warning colors")

As for the "signing blob": Easy to get rid of that.

```gradle
android {
    dependenciesInfo {
        // Disables dependency metadata when building APKs.
        includeInApk = false
        // Disables dependency metadata when building Android App Bundles.
        includeInBundle = false
    }
}
```

For some background: that BLOB is supposed to be just a binary representation of your app's dependency tree. But as it's encrypted with a public key belonging to Google, only Google can read it – and nobody else can even verify what it contains.

**Comment by IzzySoft** [comment]
:partying_face: Thanks!

**Comment by IzzySoft** [comment]
If you want more of that background: https://gitlab.com/IzzyOnDroid/repo/-/issues/475 :wink: Will probably write an article on the whole thing soon.

**Comment by IzzySoft** [comment]
That would be a nice idea indeed. But Fastlane doesn't specify anything for that, and I'd rather not put things in there and trust everything keeps working.

Also, these things are currently part of my scanning definitions (showing the explanations is a rather nice side-effect of the process, which originally was meant as "internal notes why a given permission is fine" which i wanted to make transparent), so I'd rather have control of what's there. I'd need to separate scanning "markers" from what's shown first – and I very much doubt I find time for that in the near feature. Remember that my repo is more or less a "one-man-show".

**TL;DR:** sorry, but for now you'd have to ping me if you want something updated there.
